src/dmd/root/dcompat.h

```diff
@@ -10,11 +10,24 @@
 #pragma once
 
 #include <stddef.h>
+#include <string.h>
 
 /// Represents a D [ ] array
 template<typename T>
 struct DArray
 {
     size_t length;
     T *ptr;
+
+    DArray(size_t length, T *ptr)
+    {
+        this->length = length;
+        this->ptr = ptr;
+    }
+
+    DArray(T *ptr)
```

Comment:
This only works if T is a char type.

Reply:
Right, I'll remove that then, and will introduce a DString in another PR.